services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -1225,6 +1225,12 @@ private void updateStoreVersionStatus(String storeName, int versionNumber, Versi
                 store.getName(),
                 versionNumber,
                 regionName);
+
+            // For target region push w/ deferred version swap, mark status as PUSHED as it will be marked ONLINE after
+            // roll forward
+            if (!isDeferredSwap) {
```

Comment:
Can you please add a log here or modify the previous log, also check for explicit deferred swap in non-target region configs. also update the comment to indicate it happens only in non-target region.